File: java/src/main/java/cn/az/code/other/array/SingleNumber2.java
package cn.az.code.other.array;

/**
 * @author az
 * @since 2020-06-22 21:16
 */
public class SingleNumber2 {

    public int singleNumber(int[] nums) {
        int res = 0;
        for (int i = 0; i < 32; i++) {
            int sum = 0;
            for (int n : nums) {
                if ((n >> i & 1) == 1) {
                    sum++;
                }
            }
            sum %= 3;
            res |= sum << i;
        }
        return res;
    }
}

File: archive/go/practices/array/single-number-2.go
package array

func singleNumber2(nums []int) int {
	res := 0
	for i := 0; i < 32; i++ {
		sum := 0
		for _, n := range nums {
			if (n >> i & 1) == 1 {
				sum++
			}
		}
		sum %= 3
		res |= sum << i
	}
	return res
}
